src/schemathesis/specs/openapi/expressions/parser.py

```diff
@@ -0,0 +1,90 @@
+from functools import lru_cache
+from typing import Generator, List, Union
+
+from . import lexer, nodes
+from .errors import RuntimeExpressionError, UnknownToken
+
+
+@lru_cache()  # pragma: no mutate
+def parse(expr: str) -> List[nodes.Node]:
+    """Parse lexical tokens into concrete expression nodes."""
+    return list(_parse(expr))
+
+
+def _parse(expr: str) -> Generator[nodes.Node, None, None]:
+    # pylint: disable=too-many-branches
+    tokens = lexer.tokenize(expr)
+    brackets_stack: List[str] = []
+    for token in tokens:
+        if token.is_string:
+            yield nodes.String(token.value)
+        elif token.is_variable:
+            if token.value == "$url":
```

Comment:
`$url` and other strings could be in a separate enum, so they are defined in a single place